homeassistant/components/matter/event.py

```diff
@@ -0,0 +1,135 @@
+"""Matter event entities from Node events."""
+from __future__ import annotations
+
+from typing import Any
+
+from chip.clusters import Objects as clusters
+from matter_server.client.models import device_types
+from matter_server.common.models import EventType, MatterNodeEvent
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+SwitchFeature = clusters.Switch.Bitmaps.SwitchFeature
+
+EVENT_TYPES_MAP = {
+    # mapping from raw event id's to translation keys
+    0: "switch_latched",  # clusters.Switch.Events.SwitchLatched
+    1: "initial_press",  # clusters.Switch.Events.InitialPress
+    2: "long_press_ongoing",  # clusters.Switch.Events.LongPress
+    3: "short_release",  # clusters.Switch.Events.ShortRelease
+    4: "long_release",  # clusters.Switch.Events.LongRelease
+    5: "multi_press_ongoing",  # clusters.Switch.Events.MultiPressOngoing
+    6: "multi_press_complete",  # clusters.Switch.Events.MultiPressComplete
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter switches from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.EVENT, async_add_entities)
+
+
+class MatterEventEntity(MatterEntity, EventEntity):
+    """Representation of a Matter Event entity."""
+
+    _attr_translation_key = "push"
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the entity."""
+        super().__init__(*args, **kwargs)
+        # fill the event types based on the features the switch supports
+        event_types: list[str] = []
+        feature_map = int(
+            self.get_matter_attribute_value(clusters.Switch.Attributes.FeatureMap)
+        )
+        if feature_map & SwitchFeature.kLatchingSwitch:
+            event_types.append("switch_latched")
+        if feature_map & SwitchFeature.kMomentarySwitch:
```

Comment:
Maybe rewrite to something like: 
```
if feature_map := int(...):
  if SwitchFeature.kLatchingSwitch:
    event_types.append(...)
  if SwitchFeature.kMomentarySwitch:
    event_types.append(...)
```

Not sure if adding a switch (i.e. match/case (py310+)) is recommended within HA (would opt for better reading in this case

Reply:
uhh this is a bitwise operation

Reply:
Switching programming languages more than twice a day isn't proper. Apologies, disregard